src/domain/spaces/address-books/address-book-items.repository.integration.spec.ts

```diff
@@ -0,0 +1,299 @@
+      const { spaceId, authPayload } = await createSpaceAsAdmin();
+      const items = range(1, 5).map(() => ({
+        chainIds: range(2, 5).map(() => faker.string.numeric()),
+        address: getAddress(faker.finance.ethereumAddress()),
+        name: nameBuilder(),
+        space: { id: spaceId },
+        createdBy: getAddress(faker.finance.ethereumAddress()),
+        lastUpdatedBy: getAddress(faker.finance.ethereumAddress()),
+      }));
+      await Promise.all(
+        items.map((item) => dbAddressBookItemsRepository.insert(item)),
+      );
+
+      const result = await addressBookItemsRepository.findAllBySpaceId({
+        authPayload,
+        spaceId,
+      });
+
+      expect(result).toHaveLength(items.length);
+      expect(result).toEqual(
+        expect.arrayContaining(
+          items.map((item) =>
+            expect.objectContaining({
+              chainIds: item.chainIds,
+              address: item.address,
+              name: item.name,
+              createdBy: item.createdBy,
+              lastUpdatedBy: item.lastUpdatedBy,
+            }),
+          ),
+        ),
+      );
+    });
+
+    it('should return an array of address book items for the given Space ID for a Space Member', async () => {
+      const { spaceId } = await createSpaceAsAdmin();
+      const authPayload = await addMemberToSpaceWithStatus(spaceId, 'ACTIVE');
+      const items = range(1, 5).map(() => ({
+        chainIds: range(2, 5).map(() => faker.string.numeric()),
+        address: getAddress(faker.finance.ethereumAddress()),
+        name: nameBuilder(),
+        space: { id: spaceId },
+        createdBy: getAddress(faker.finance.ethereumAddress()),
+        lastUpdatedBy: getAddress(faker.finance.ethereumAddress()),
+      }));
+      await Promise.all(
+        items.map((item) => dbAddressBookItemsRepository.insert(item)),
+      );
+
+      const result = await addressBookItemsRepository.findAllBySpaceId({
+        authPayload,
+        spaceId,
+      });
+
+      expect(result).toHaveLength(items.length);
+      expect(result).toEqual(
+        expect.arrayContaining(
+          items.map((item) =>
+            expect.objectContaining({
+              chainIds: item.chainIds,
+              address: item.address,
+              name: item.name,
+              createdBy: item.createdBy,
+              lastUpdatedBy: item.lastUpdatedBy,
+            }),
+          ),
+        ),
+      );
+    });
+
+    it('should return a NotFoundException if the space does not exist', async () => {
+      const { authPayload } = await createUser();
+      await expect(
+        addressBookItemsRepository.findAllBySpaceId({
+          authPayload,
+          spaceId: faker.number.int({ min: 1, max: DB_MAX_SAFE_INTEGER }),
+        }),
+      ).rejects.toThrow(new NotFoundException('Space not found.'));
+    });
+
+    it('should throw an NotFoundException if the user is not a member', async () => {
```

Comment:
Nit:
```suggestion
    it('should throw a NotFoundException if the user is not a member', async () => {
```

The next test is also the same but I won't add another comment.